src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+    """
+    Upload a file, optionally compressing it.
+
+    Args:
+        filename (str): The file to upload.
+        skip_compress (bool): Whether to skip gzip compression.
+        kwargs: Additional kwargs to the upload function.
+
+    Returns:
+        A dxlink or DXFile.
+    """
+    if skip_compress:
+        return simple_upload_file(filename, **kwargs)
+    else:
+        return compress_and_upload_file(filename, **kwargs)
+
+
+def simple_upload_file(
+    filename,
+    remote_filename=None,
+    remote_folder="/",
+    return_handler=False,
+    project=None,
+    wait_on_close=False,
+):
+    """
+    Upload a file and return a link.
+
+    Args:
+        filename (str): Local filename.
+        remote_filename (str): Optional, remote filename to upload file to.
+        remote_folder (str): Optional, remote folder to upload file to.
+        return_handler (bool): Whether to return a DXFile handler.
+        project (str): The project ID to upload to, if not the currently selected
+            project.
+        wait_on_close (bool): Whether to block until the file has closed.
+
+    Returns:
+        DNAnexus link or DXFile pointing to the uploaded file (depending on
+        the value of `return_handler`).
+    """
+    kwargs = {"wait_on_close": wait_on_close}
+
+    if remote_filename:
+        kwargs["name"] = remote_filename
+    if not remote_folder:
+        remote_folder = "/"
+    elif not remote_folder.startswith("/"):
+        remote_folder = "/{}".format(remote_folder)
+    kwargs["folder"] = remote_folder
+    if project:
+        kwargs["project"] = project
+    LOG.info("Uploading file %s to %s", filename, remote_filename or filename)
+    handler = dxpy.upload_local_file(filename, **kwargs)
+    if return_handler:
+        return handler
+    else:
+        return dxpy.dxlink(handler.get_id(), handler.describe()["project"])
+
+
+def compress_and_upload_file(
+    filename,
+    remote_filename=None,
+    remote_folder="/",
+    compression_level=1,
+    return_handler=False,
+    project=None,
+    wait_on_close=False,
+    method="gz",
+    max_part_size=None
+):
+    """
+    Gzip and upload a local file.
+
+    Shorthand for running 'gzip' and 'dx upload' using a subprocess on a given
+    local file.
+
+    Args:
+        filename (str): Local filename.
+        remote_filename (str): Optional, remote filename to upload file to.
+        remote_folder (str): Optional, remote folder to upload file to.
+        compression_level (int): Level of compression between 1 and 9 to compress
+            file to. Specify 1 for gzip --fast and 9 for gzip --best. If not
+            specified, --fast is assumed.
+        return_handler (bool): Whether to return a DXFile handler.
+        project (str): The project ID to upload to, if not the currently selected
+            project.
+        wait_on_close (bool): Whether to block until the file has closed.
+        method (str): Compression method; one of 'gz', 'bz2'.
```

Comment:
Method implies executable to me, either "gzip" or "bgzip". Perhaps replace `method` with `extension` and then use the appropriate exe from there.

Reply:
Changed it to 'compression_type'